engine/class_modules/sc_shaman.cpp

```diff
@@ -5428,7 +5431,13 @@ struct primordial_wave_t : public shaman_spell_t
 
   void execute() override {
     shaman_spell_t::execute();
+
     p()->buff.primordial_wave->trigger();
+
+    if ( p()->conduit.tumbling_waves->ok() && rng().roll( p()->conduit.tumbling_waves.percent() ) )
+    {
+      cooldown->reset( false );
```

Comment:
Not entirely sure this makes a difference in this particular case, but we can set this to true and it should be fine:

```suggestion
      cooldown->reset( true );
```